adhocteam/Head-Start-TTADP
# ClamAV infrastructure updates

**Description of change**

Update diagrams and network policy rules for the new Go-based ClamAV app.

ClamAV exposes a TLS endpoint using a self-signed certificate, so it's likely that the code will need to skip SSL verification.

Once merged and applied, ClamAV will be available at `https://clamapi-ttahub-dev.apps.internal:9443` for all non-prod cloud.gov environments and `https://clamapi-ttahub-prod.apps.internal:9443` in production.

**How to test**

```
┌───────────────────>
│Head-Start-TTADP/terraform/dev on  clamav-updates [$] via 💠 default on  (us-gov-west-1) ✅ 
└─> terraform plan
cloudfoundry_service_instance.document_upload_bucket: Refreshing state... [id=5d4ab5b7-fa4e-4bb3-bca9-ea317ac6351e]
cloudfoundry_service_instance.database: Refreshing state... [id=5bc4224a-fe6f-42c3-859d-b421be20d90d]
cloudfoundry_network_policy.clamav_routing: Refreshing state... [id=4d36b2f3-59a5-9ff4-3975-006d0c50b153]

An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
  ~ update in-place

Terraform will perform the following actions:

  # cloudfoundry_network_policy.clamav_routing will be updated in-place
  ~ resource "cloudfoundry_network_policy" "clamav_routing" {
        id = "4d36b2f3-59a5-9ff4-3975-006d0c50b153"

      + policy {
          + destination_app = "84f61b43-c63e-41db-ad23-573a3d42134a"
          + port            = "9443"
          + protocol        = "tcp"
          + source_app      = "dab6ff3f-786f-443b-bfb2-4e862b5ad33e"
        }
    }

Plan: 0 to add, 1 to change, 0 to destroy.

------------------------------------------------------------------------
```

**Issue(s)**
* https://github.com/HHS/Head-Start-TTADP/issues/298

**Checklist**

    - API methods
    - Boundary and Data Flow Diagrams
    - [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions with the [Nygard template](https://github.com/joelparkerhenderson/architecture_decision_record/blob/master/adr_template_by_michael_nygard.md)
    - OSCAL templates completed when security controls are implemented or modified
